server/src/leihs/procurement/resources/viewers.clj

```diff
@@ -1,38 +1,43 @@
 (ns leihs.procurement.resources.viewers
-  (:require [clojure.java.jdbc :as jdbc]
-            [clojure.tools.logging :as log]
-            [leihs.procurement.resources.users :refer [users-base-query]]
-            [leihs.procurement.utils.sql :as sql]))
+  (:require
+    [honey.sql :refer [format] :rename {format sql-format}]
+    [honey.sql.helpers :as sql]
+    [leihs.core.utils :refer [my-cast]]
+    [leihs.procurement.resources.users :refer [users-base-query]]
+    [next.jdbc :as jdbc]
+    [taoensso.timbre :refer [debug error info spy warn]]))
 
 (defn get-viewers
   [context _ value]
-  (jdbc/query (-> context
-                  :request
-                  :tx)
-              (-> users-base-query
-                  (sql/merge-where
-                    [:in :users.id
-                     (-> (sql/select :pcv.user_id)
-                         (sql/from [:procurement_category_viewers :pcv])
-                         (sql/merge-where [:= :pcv.category_id (:id value)]))])
-                  sql/format)))
+  (jdbc/execute! (-> context
+                     :request
+                     :tx-next)
+                 (-> users-base-query (sql/where [:in :users.id
+                                                  (-> (sql/select :pcv.user_id)
+                                                      (sql/from [:procurement_category_viewers :pcv])
+                                                      (sql/where [:= :pcv.category_id [:cast (:id value) :uuid]]))])
+                     sql-format)))
 
 (defn delete-viewers-for-category-id!
   [tx c-id]
-  (jdbc/execute! tx
-                 (-> (sql/delete-from [:procurement_category_viewers :pcv])
-                     (sql/merge-where [:= :pcv.category_id c-id])
-                     sql/format)))
+  (let [result (jdbc/execute-one! tx (-> (sql/delete-from :procurement_category_viewers :pcv)
+                                              (sql/where [:= :pcv.category_id [:cast c-id :uuid]])
+                                              sql-format))
+
+        res (spy (:update-count result))                    ;; TODO: fixme
+        result (spy (:next.jdbc/update-count result))]
```

Comment:
jdbc-next prefix needed

Reply:
in this case the prefix has something to do with the internals of `next.jdbc` and i am afraid there is not much that can be done about it. but generally we are not interested in the `update-count` so one can leave it like it is now on `master`, just do the side-effect (`UPDATE`, `INSERT`) and don't care about the return value.